Transaction failure when new node lacks a property in the index spec

My index spec reads `elasticsearch.index_spec=stmpd:stmpd(notes)`. Some of my stmpd-labeled nodes have a `notes` field and some do not. When I use the plugin, if I try to create a new node without the `notes` field, I get a Neo4j error:

>Transaction was marked as successful, but unable to commit transaction so rolled back.

Is this intended behavior? If it is, is there some way that I can configure the plugin to allow the creation of new Neo4j nodes with an optional indexed property?

Just created a PR, don't know whether it's acceptable but happy to incorporate feedback. Also, @jazzido I tried building your fork but it failed on my machine (I'm on OS X with Java 8 Update 31). I'm posting what I think is the relevant part of the backtrace:

>Running org.neo4j.elasticsearch.ElasticSearchImportAppTest
Welcome to the Neo4j Shell! Enter 'help' for a list of commands

>Connected to ES cluster: http://localhost:9200
Welcome to the Neo4j Shell! Enter 'help' for a list of commands

>Connected to ES cluster: http://localhost:9200
Indexing MyLabel to my_index
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.901 sec <<< FAILURE!
testBadSyntaxShouldThrow(org.neo4j.elasticsearch.ElasticSearchImportAppTest)  Time elapsed: 0.251 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: java.lang.Exception
        at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:32)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
        at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
        at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)

--- a/src/main/java/org/neo4j/elasticsearch/ElasticSearchEventHandler.java
+++ b/src/main/java/org/neo4j/elasticsearch/ElasticSearchEventHandler.java
@@ -104,6 +104,12 @@ class ElasticSearchEventHandler implements TransactionEventHandler<Collection<Bu
     private boolean hasLabel(PropertyEntry<Node> propEntry) {
         return hasLabel(propEntry.entity());
     }
+
+    private Set<String> getIntersectingProperties(ElasticSearchIndexSpec spec, Node node) {
+      Set<String> specProperties = new HashSet<String>(spec.getProperties());
+      specProperties.retainAll((Collection)node.getPropertyKeys());
+      return specProperties;
+    }
     
     private Map<IndexId, Index> indexRequests(Node node) {
         HashMap<IndexId, Index> reqs = new HashMap<>();
@@ -113,7 +119,8 @@ class ElasticSearchEventHandler implements TransactionEventHandler<Collection<Bu
 
             for (ElasticSearchIndexSpec spec: indexSpecs.get(l)) {
                 String id = id(node), indexName = spec.getIndexName();
-                reqs.put(new IndexId(indexName, id), new Index.Builder(nodeToJson(node, spec.getProperties()))
+                Set<String> intersectingProperties = getIntersectingProperties(spec, node);
+                reqs.put(new IndexId(indexName, id), new Index.Builder(nodeToJson(node, intersectingProperties))
                 .type(l.name())
                 .index(indexName)
                 .id(id)
@@ -161,8 +168,9 @@ class ElasticSearchEventHandler implements TransactionEventHandler<Collection<Bu
 
     		for (ElasticSearchIndexSpec spec: indexSpecs.get(l)) {
     		    String id = id(node), indexName = spec.getIndexName();
+            Set<String> intersectingProperties = getIntersectingProperties(spec, node);
     			reqs.put(new IndexId(indexName, id),
-    			        new Update.Builder(nodeToJson(node, spec.getProperties()))
+    			        new Update.Builder(nodeToJson(node, intersectingProperties))
                     			  .type(l.name())
                     			  .index(spec.getIndexName())
                     			  .id(id(node))

